sdk/eventhub/event-hubs/src/eventHubClient.ts

```diff
@@ -171,58 +178,85 @@ export class EventHubClient {
     options?: ClientOptions
   );
   constructor(
-    host: string,
-    entityPath: string,
-    tokenProviderOrCredentials:
+    hostOrConnectionString: string,
+    eventHubPathOrOptions?: string | ClientOptions,
+    tokenProviderOrCredentials?:
       | TokenProvider
       | ApplicationTokenCredentials
       | UserTokenCredentials
       | DeviceTokenCredentials
       | MSITokenCredentials,
     options?: ClientOptions
   ) {
-    if (!host) {
-      throw new Error(
-        "Please provide a fully qualified domain name of your Event Hub instance for the 'host' parameter."
+    let connectionString;
+    let sharedAccessKeyName = "defaultKeyName";
+    let sharedAccessKey = "defaultKeyValue";
+    let eventHubPath: string | undefined;
+    let tokenProvider = {} as TokenProvider;
+
+    hostOrConnectionString = String(hostOrConnectionString);
+
+    if (hostOrConnectionString.startsWith("Endpoint")) {
+      connectionString = hostOrConnectionString;
+      const parsedCS = parseConnectionString<ServiceBusConnectionStringModel>(
+        connectionString
       );
+      if (!parsedCS.EntityPath){
+        throw new TypeError(
+          `"connectionString": "${connectionString}", ` +
+            `must contain EntityPath="<path-to-the-entity>".`
+        );
+      }
+    } else {
+      let host = hostOrConnectionString;
+      if (!host) {
+        throw new Error(
+          "Please provide a fully qualified domain name of your Event Hub instance for the 'host' parameter."
+        );
+      }
+      if (!host.endsWith("/")) host += "/";
+      connectionString = `Endpoint=sb://${host};SharedAccessKeyName=${sharedAccessKeyName};SharedAccessKey=${sharedAccessKey};`;
     }
 
-    if (!entityPath) {
-      throw new Error("Please provide a value for the 'entityPath' parameter.");
-    }
+    if (typeof eventHubPathOrOptions === "string") {
+      eventHubPath = eventHubPathOrOptions;
+      if (!eventHubPath) {
+        throw new Error("Please provide a value for the 'entityPath' parameter.");
+      }
+      if (!tokenProviderOrCredentials) {
+        throw new Error("Please provide either a token provider or a valid credentials object.");
+      }
 
-    if (!tokenProviderOrCredentials) {
-      throw new Error("Please provide either a token provider or a valid credentials object.");
+      if (
+        tokenProviderOrCredentials instanceof ApplicationTokenCredentials ||
+        tokenProviderOrCredentials instanceof UserTokenCredentials ||
+        tokenProviderOrCredentials instanceof DeviceTokenCredentials ||
+        tokenProviderOrCredentials instanceof MSITokenCredentials
+      ) {
+        tokenProvider = new AadTokenProvider(tokenProviderOrCredentials);
+      } else {
+        tokenProvider = tokenProviderOrCredentials;
+        if (tokenProvider instanceof SasTokenProvider) {
+          sharedAccessKeyName = tokenProvider.keyName;
+          sharedAccessKey = tokenProvider.key;
```

Comment:
`sharedAccessKeyName` and `sharedAccessKey` get updated here but the updated value doesnt get used. This is because, you moved the creation of the connection string before this point :)